### DISCUSSION TITLE
CDN for "vanilla Javascript" ?

### DISCUSSION THREAD

**Discussion by fbaube** [ueberdosis/tiptap#2529] [General Questions]
Regarding the discussion at https://tiptap.dev/installation#vanilla-java-script

This apparently loads Javascript from the local npm installation. The mechanism for getting the Javascript into the browser Javascript runtime is not obvious and it seems like it might not be robust. 

Is it possible to explicitly load all the TipTap Javascript (and other resources) over the internet using script tags in the HTML header ? Many other in-browser editors can do this. 

My apologies if this is a dumb question.

**Comment by philippkuehn** [comment]
There was a CDN example but there are some issues with skypack and esm.sh so we decided to remove it from the docs for now.
See: https://github.com/ueberdosis/tiptap/issues/1537

**Comment by ricotrevisan** [comment]
In the meantime, the CDN version has been re-added to the docs (I think I'm guilty of that PR). And the CDN version is again having issues. It loads normally, but it breaks when you hit Enter at the end of the text.

FYI: I already posted this on [esm.sh's repo](https://github.com/esm-dev/esm.sh/discussions/845)  hoping @ije can do some magic. I'm wondering if a clean solution would be updating Tiptap's repo -- maybe the pinning the [`prosemirror-model` version](https://github.com/ueberdosis/tiptap/blob/aba6657d7026cf59a41b9532b767c8e3058b1b03/packages/pm/package.json#L139)?

---

# Error in console
![CleanShot 2024-06-15 at 16 29 45@2x](https://github.com/esm-dev/esm.sh/assets/2459327/c1e17d0e-f95d-4c37-82b7-418ee2481d07)


# Network tab
Looking at the network tab, I can see that `prosemirror-model` was indeed loaded multiple times

![CleanShot 2024-06-15 at 16 31 16@2x](https://github.com/esm-dev/esm.sh/assets/2459327/d6983012-4997-4c5a-9070-8aef000ff5a5)


# Inspecting
And they load 2 different versions

![CleanShot 2024-06-15 at 16 36 14@2x](https://github.com/esm-dev/esm.sh/assets/2459327/3c4d8e1f-f7ca-4a21-81f1-0ee5ce66a62c)


---

# Steps to reproduce
1. open this Codepen: https://codepen.io/ricotrevisan/pen/poqambQ
2. open the browser's console (not Codepen's console)
3. reload the page
4. in the page, after "Example Text", hit enter
5. the error shows up in the browser's console (not Codepen's console).

# What I've tried
- I've tried esm.sh and skypack.dev: same problem
- with esm.sh, I've tried specifying:
  - version of dependencies
  - version of the tiptap-extension
  - version of esm.sh
  - ... same issue with different version numbers.

**Comment by ije** [comment]
i don't think you can it now. you can use import maps to make the dependency as an exteranl module, check https://preactjs.com/guide/v10/getting-started#using-preact-with-htm-and-importmaps
next version fo esm.sh going to use semver caret(e.g. ^1.0.0) range for dependecies instead of current fixed version, should imrove the duplicate issue
